osx-setup/make-release.sh

```diff
@@ -3,60 +3,72 @@ VERSION=$1
-echo "Fixing CFBundleIdentifier and CFBundleIconFile"
+
+echo "Fixing CFBundleIdentifier and CFBundleIconFile..."
 # MacOS has to be different, ugh...
 # https://stackoverflow.com/questions/32004950/how-to-replace-string-in-a-file-in-place-using-sed
 sed -i '' s/com.yourcompany.lokicp/network.loki.lokinet/ osx-pkg/Applications/Lokinet.app/Contents/Info.plist
-sed -i '' s/<string></string>/<string>lokinet.icns</string> osx-pkg/Applications/Lokinet.app/Contents/Info.plist
-
-echo "Generating distribution version"
-sed "s/\$VERSION/$VERSION/" distribution_template.xml > distribution.xml
+sed -i '' "s/\<string\>\<\/string\>/\<string\>lokinet.icns\<\/string\>/" osx-pkg/Applications/Lokinet.app/Contents/Info.plist
 
-echo "Signing package $VERSION"
-codesign --options=runtime -s "Rangeproof PTY LTD" -v osx-pkg/usr/local/bin/lokinet
-codesign --options=runtime -s "Rangeproof PTY LTD" -v osx-pkg/usr/local/bin/lokinetctl
-codesign --options=runtime -s "Rangeproof PTY LTD" -v osx-pkg/usr/local/bin/lokinet-bootstrap
-codesign --options=runtime -s "Rangeproof PTY LTD" -v osx-pkg/usr/local/lib/libuv.dylib
+echo "Signing package $VERSION..."
 codesign --deep --options=runtime -s "Rangeproof PTY LTD" -v osx-pkg/Applications/Lokinet.app
 
-echo "Building package $VERSION"
-mkdir -p pkg1
-rm pkg1/lokinet.pkg
+echo "Building main component package $VERSION..."
 pkgbuild --root osx-pkg --scripts scripts --identifier network.loki.lokinet --version $VERSION pkg1/lokinet.pkg
-rm lokinet-v$VERSION.pkg
+
+echo "Setting up dtd-pkg"
+mkdir -p dtd-pkg/Applications/Lokinet.app/Contents/Resources
+touch dtd-pkg/Applications/Lokinet.app/Contents/Resources/disable_auto_dns
+
+# I don't think we need to sign a 0 byte file
+echo "Building dtd component package..."
+pkgbuild --root dtd-pkg --identifier network.loki.lokinet_dtd --version $VERSION pkg1/lokinet-dtd.pkg
+
+echo "Generating distribution version..."
+sed "s/\$VERSION/$VERSION/" distribution_template.xml > distribution.xml
+
+echo "Assembling installer..."
 productbuild --distribution distribution.xml --resources resources --package-path pkg1 --version $VERSION lokinet_macos64_installer_v$VERSION.pkg
+echo "Signing .pkg..."
 productsign --sign F87AE1FB2AEDAE8A15F9510B6DD1AB2B04E384A1 lokinet_macos64_installer_v$VERSION.pkg lokinet_macos64_installer_v$VERSION-signed.pkg
 #pkgutil --check-signature lokinet_macos64_installer_v$VERSION-signed.pkg
+
+echo "Configuring gatekeeper..."
 spctl -a -v --type install lokinet_macos64_installer_v$VERSION-signed.pkg
+
+echo "Submitting Notarization..."
 xcrun altool --notarize-app --primary-bundle-id "network.loki.loki-network.pkg" --username $EMAIL --password $PW --asc-provider "SUQ8J2PCT7" --file lokinet_macos64_installer_v$VERSION-signed.pkg
-echo "xcrun altool --notarization-info X --username $EMAIL --password $PW"
+
+echo "xcrun altool --notarization-info XXXXXXXXX --username $EMAIL --password $PW"
```

Comment:
why are we echoing secrets to stdout?

Reply:
macOS a shit perhaps? even `signtool(1)` doesn't do this

Reply:
It’s a parameter passed in, so it’s already in the console. It’s just echo out shortcuts of commands you have to run manually after (I haven’t been able to programmatically capture the output and parse it yet)